pkgs/by-name/ko/komac/package.nix

```diff
@@ -7,32 +7,41 @@
 , darwin
 , testers
 , komac
+, zstd
 }:
 
 let
-  version = "2.2.1";
+  version = "2.3.0";
   src = fetchFromGitHub {
     owner = "russellbanks";
     repo = "Komac";
     rev = "v${version}";
-    hash = "sha256-dPX8/JUQ+vugd+M/jIjBf4/sNbac0FVQ0obhyAAGI84=";
+    hash = "sha256-swl2WEFE2Jglo76KseDlb5K8vQ7DCMcLlxAFhdqTv7I=";
   };
 in
 rustPlatform.buildRustPackage {
   inherit version src;
 
   pname = "komac";
 
-  cargoHash = "sha256-CDPN90X3m/9FRLolAVCIcAuajZbB5OAgLcFXq2ICS8g=";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
>Use cargoLock.lockFile instead of cargoHash because using cargoHash returns this error:

https://github.com/NixOS/nixpkgs/blob/13da3c09fb023da02ff4752096e7854338d141d4/doc/languages-frameworks/rust.section.md?plain=1#L55-L64

In this Rust packaging guide, there are both cargoHash and Cargo.lock examples.
Which way is desirable?

https://github.com/NixOS/nixpkgs/pull/296350

>No git dependency in the Cargo.lock https://github.com/russellbanks/Komac/blob/f0bce16d8dd8d78fb5ea0bdcb45a64831cef7470/Cargo.lock

I was thinking we can simply updating the cargoHash field. 🤔

Reply:
I couldn't make it build with cargoHash.